### PR TITLE
FavouriteDragon/a5.0 cloudburst levels merge

### PR DESCRIPTION
**Pull request by FavouriteDragon**
Yay we can merge them

### CODE REVIEW

File: src/main/java/com/crowsofwar/avatar/common/entity/EntityCloudBall.java [hunk 1]
```diff
@@ -54,12 +65,11 @@ public void entityInit() {
     @Override
     public void onUpdate() {
         super.onUpdate();
+        CloudburstBehavior.Thrown thrown = new CloudburstBehavior.Thrown();
         setBehavior((CloudburstBehavior) getBehavior().onUpdate(this));
-        if (ticksExisted >= 250) {
+        if (ticksExisted >= 250 && this.getBehavior() == thrown) {
```
Comment (CrowsOfWar) [inline src/main/java/com/crowsofwar/avatar/common/entity/EntityCloudBall.java]: Change `this.getBehavior() == thrown` to `getBehavior() instanceof CloudBurstBehavior.Thrown`

File: src/main/java/com/crowsofwar/avatar/common/entity/EntityCloudBall.java [hunk 2]
```diff
@@ -54,12 +65,11 @@ public void entityInit() {
     @Override
     public void onUpdate() {
         super.onUpdate();
+        CloudburstBehavior.Thrown thrown = new CloudburstBehavior.Thrown();
```
Comment (CrowsOfWar) [inline src/main/java/com/crowsofwar/avatar/common/entity/EntityCloudBall.java]: This isn't necessary, Please see the line below for why not.

File: src/main/java/com/crowsofwar/avatar/common/entity/EntityCloudBall.java [hunk 3]
```diff
@@ -119,6 +129,58 @@ public boolean onCollideWithSolid() {
...
+                    return false;
+                }
+            }
+            if (chismash){
+                if (entity instanceof EntityLivingBase) {
+                    if (Bender.isBenderSupported((EntityLivingBase) entity)) {
+                        BendingData data = BendingData.get((EntityLivingBase) entity);
+                        CloudburstPowerModifier cloudModifier = new CloudburstPowerModifier();
+                        cloudModifier.setTicks(100);
+                        data.getPowerRatingManager(Airbending.ID).addModifier(new CloudburstPowerModifier());
```
Comment (CrowsOfWar) [inline src/main/java/com/crowsofwar/avatar/common/entity/EntityCloudBall.java]: To affect all bending controllers I would do this

```
For (UUID uuid : data.getBendingIds()) {
data.getPowerRatingManager(uuid). Add modifier
}
```

Comment (FavouriteDragon) [reply src/main/java/com/crowsofwar/avatar/common/entity/EntityCloudBall.java]: ah ok thanks!

File: src/main/java/com/crowsofwar/avatar/common/entity/EntityCloudBall.java (hunk 1 above)
Comment (FavouriteDragon) [reply src/main/java/com/crowsofwar/avatar/common/entity/EntityCloudBall.java]: oohhhh
ok I'll do that!
thanks!

File: src/main/java/com/crowsofwar/avatar/common/entity/EntityCloudBall.java (hunk 3 above)
Comment (FavouriteDragon) [reply src/main/java/com/crowsofwar/avatar/common/entity/EntityCloudBall.java]: actully there is no getBendingIds/getBendingID
should I getActiveBending?

Comment (FavouriteDragon) [reply src/main/java/com/crowsofwar/avatar/common/entity/EntityCloudBall.java]: use getActiveBending

Comment (FavouriteDragon) [reply src/main/java/com/crowsofwar/avatar/common/entity/EntityCloudBall.java]: yeah it doesn't like this:
 for (UUID uuid : data.getActiveBendingId()) {
                            data.getPowerRatingManager(uuid).addModifier(cloudModifier);
it complains about how i defined UUID

File: src/main/java/com/crowsofwar/avatar/common/entity/EntityCloudBall.java [hunk 4]
```diff
@@ -161,10 +162,11 @@ protected boolean canCollideWith(Entity entity) {
                     if (Bender.isBenderSupported((EntityLivingBase) entity)) {
                         BendingData data = BendingData.get((EntityLivingBase) entity);
                         CloudburstPowerModifier cloudModifier = new CloudburstPowerModifier();
```
Comment (CrowsOfWar) [inline src/main/java/com/crowsofwar/avatar/common/entity/EntityCloudBall.java]: Would you be able to move this to the beginning of the for loop?

**Pull request merged**